File: app/src/main/java/com/google/fhirengine/MainActivity.java
package com.google.fhirengine;

import android.app.Activity;
import android.os.Bundle;

public class MainActivity extends Activity {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    // Gets FHIR Engine using Dagger component.
    FhirEngine fhirEngine = DaggerFhirEngineComponent.builder()
        .context(this).build().getFhirEngine();
  }
}

File: codelabs/datacapture/app/src/main/java/com/google/codelab/sdclibrary/MainActivity.kt
/*
 * Copyright 2022-2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.google.codelab.sdclibrary

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

  var questionnaireJsonString: String? = null

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)

    // 4.2 Replace with code from the codelab to add a questionnaire fragment.
  }

  private fun submitQuestionnaire() =
    lifecycleScope.launch {
      // 5 Replace with code from the codelab to get a questionnaire response.

      // 6 Replace with code from the codelab to extract FHIR resources from QuestionnaireResponse.
    }

  private fun getStringFromAssets(fileName: String): String {
    return assets.open(fileName).bufferedReader().use { it.readText() }
  }
}
